### PR TITLE
Release: 4.0.0

### PR DESCRIPTION
**Pull request by aboedo**
## 4.0.0

RevenueCat Unity SDK v4 is here!! 

Dancing cats

[Full Changelog](https://github.com/revenuecat/purchases-unity/compare/3.5.3...4.0.0)

#### Amazon Appstore Support
We have introduced support for using the Amazon Appstore. We have extensively tested this, and there are some apps using our pre-release Amazon versions in production.

However, we have found some inconsistencies in the way Amazon Appstore prices are reported. We are actively working on patching these inconsistencies.

Please help us help you by reporting any issues you find. [New RevenueCat Issue](https://github.com/RevenueCat/purchases-unity/issues/new/).

You can enable Amazon Appstore support by configuring the SDK using the new `RevenueCatAPIKeyAmazon` field.

For more information around configuration please take a look at the [Amazon Appstore section in our docs](https://docs.revenuecat.com/docs/amazon-platform-resources). The official [Amazon In-App Purchasing docs](https://developer.amazon.com/docs/in-app-purchasing/iap-overview.html) also contain very valuable information, especially around testing and best practices.

⚠️ ⚠️ Important ⚠️ ⚠️ In order to use Unity IAP in order with RevenueCat in Observer mode,
you need to use the `Purchases-UnityIAP.unityPackage`.

For Amazon installation instructions please follow take a look at the [Amazon instructions document](https://github.com/RevenueCat/purchases-unity/blob/4.0.0-rc.1/AMAZON-INSTRUCTIONS.md)

#### StoreKit 2 support
This version of the SDK automatically uses StoreKit 2 APIs under the hood only for APIs that the RevenueCat team has determined work better than StoreKit 1. 

#### New types and cleaned up naming
New types that wrap native types from Apple, Google and Amazon, and we cleaned up the naming of other types and methods for a more consistent experience. 

### Removed APIs
- `Identify` and `CreateAlias` have been removed in favor of `LogIn`.
- `Reset` has been removed in favor of `LogOut`.
- `GetEntitlements` has been removed in favor of `GetOfferings`.
- `AttributionKey` and `Purchases.AddAttributionData` have been removed in favor of `Set<NetworkID> methods`.
- `revenueCatAPIKey` has been removed in favor of `revenueCatAPIKeyApple`, `revenueCatAPIKeyGoogle` and `revenueCatAPIKeyAmazon`.

### Renamed APIs

| 3.x | 4.0.0 |
| :-: | :-: |
| `PurchaserInfo` | `CustomerInfo` |
| `Transaction` | `StoreTransaction` |
| `Product` | `StoreProduct` |
| `PaymentDiscount` | `PromotionalOffer` |
| `Purchases.RestoreTransactions` | `Purchases.RestorePurchases` |
| `Purchases.GetPaymentDiscount` | `Purchases.GetPromotionalOffer` |
| `Purchases.UpdatedPurchaserInfoListener` | `Purchases.UpdatedCustomerInfoListener` |
| `Discount.identifier` | `Discount.Identifier` |
| `Discount.price` | `Discount.Price` |
| `Discount.priceString` | `Discount.PriceString` |
| `Discount.cycles` | `Discount.Cycles` |
| `Discount.period` | `Discount.Period` |
| `Discount.unit` | `Discount.Unit` |
| `Discount.periodUnit` | `Discount.PeriodUnit` |
| `Discount.periodNumberOfUnits` | `Discount.PeriodNumberOfUnits` |
| `Error.message` | `Error.Message` |
| `Error.code` | `Error.Code` |
| `Error.underlyingErrorMessage` | `Error.UnderlyingErrorMessage` |
| `Error.readableErrorCode` | `Error.ReadableErrorCode` |

### New APIs
- Introduced the ability to configure the SDK programmatically through calling `Configure` https://github.com/RevenueCat/purchases-unity/pull/125

### Other
- Added docstrings for most public entities. https://github.com/RevenueCat/purchases-unity/pull/131, https://github.com/RevenueCat/purchases-unity/pull/130, https://github.com/RevenueCat/purchases-unity/pull/128

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,71 @@
-## 4.0.0-rc.3
+## 4.0.0
 
-- Deprecated `Purchases.Setup` in favor of `Purchases.Configure`.
+RevenueCat Unity SDK v4 is here!! 
+
+![Dancing cats](https://media.giphy.com/media/lkbNG2zqzHZUA/giphy.gif)
```
Comment (NachoSoto) [inline CHANGELOG.md:5]: 🐱

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,71 @@
-## 4.0.0-rc.3
```
Comment (NachoSoto) [inline CHANGELOG.md:1]: I was going to ask why this went away, but I guess RC3 never shipped right?

**Review by NachoSoto — APPROVED** [review]
🚢

File: CHANGELOG.md (hunk 1 above)
Comment (taquitos) [reply CHANGELOG.md:5]: image

File: CHANGELOG.md (hunk 2 above)
Comment (aboedo) [reply CHANGELOG.md:1]: yeah, I was originally gonna ship this as RC3 but as I gained confidence in the build I ended up removing the prerelease modifier

**Review by tonidero — APPROVED** [review]
Looks great!

**Pull request merged**